### DISCUSSION TITLE
[Homebrew] dotnet formulae deprecated due to SSL 1.1 dependency

### DISCUSSION THREAD

**Discussion by chenrui333** [dotnet/source-build#3691] [General]
just heads up, homebrew has deprecated the dotnet formulae due to SSL 1.1 dependency. 

relates to https://github.com/Homebrew/homebrew-core/pull/152278

---

Also is this comment still relevant?

> .NET Source-Build is supported on the oldest available .NET SDK feature update for each major release, and on Linux only. For example, if .NET 6.0.1xx, 6.0.2xx, 7.0.1xx, and 7.0.2xx feature updates are available from [dotnet.microsoft.com](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), Source-Build will support 6.0.1xx and 7.0.1xx.

cc @asbjornu

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by chenrui333** [comment]
duplicate of https://github.com/dotnet/installer/issues/17624 (not sure where is a better place to file the issue)

**Comment by MichaelSimons** [comment]
cc @richlander, @ashnaga

**Comment by omajid** [comment]
.NET (including source-build) versions 6.0and later support using OpenSSL 3.0 out of the box. Are there any known issues with updating the .NET formulae to use OpenSSL 3.0 instead of 1.1?

**Comment by omajid** [comment]
> Also is this comment still relevant?
>
>> .NET Source-Build is supported on the oldest available .NET SDK feature update for each major release, and on Linux only. For example, if .NET 6.0.1xx, 6.0.2xx, 7.0.1xx, and 7.0.2xx feature updates are available from [dotnet.microsoft.com](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), Source-Build will support 6.0.1xx and 7.0.1xx.

Yes, this still mostly correct.

Only the oldest feature band is supported for .NET 6, 7 and 8. There's some plans being made to try and support multiple feature bands for .NET 9: https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Managing-SDK-Bands.md

Source-build is fully supported on Linux. There's some work to support it on macOS (for example, the Homebrew formulae) and community contributions are more than welcome. Full Windows support is being planned for .NET 9.

**Comment by asbjornu** [comment]
I suppose it shouldn't be more difficult than changing [`dotnet.rb:27`](https://github.com/Homebrew/homebrew-core/blob/67784bf7ffcffc775299bae0253121a7d9e2ee9d/Formula/d/dotnet.rb#L27) from: 

```rb
  depends_on "openssl@1.1"
```

to:

```rb
  depends_on "openssl@3.0"
```

or perhaps even:

```rb
  depends_on "openssl@3.1"
```


However, I unfortunately don't have time to do it currently. If anyone else can submit a PR, I'm sure it's going to be merged if it turns green.

**Comment by chenrui333** [comment]
The Homebrew core team does not have any expertise in the dotnet and mono systems, so we just deprecate all the dotnet/mono formulae, need some help from the community to help driving it.

**Comment by alebcay** [comment]
I think we had some issues getting `dotnet` to work with `openssl@3` in our CI, but looking back now, I think it might have been some issues with our CI. I'm still fiddling a bit with this on my local machine.

**Reply by chenrui333** [reply]
related PR on this, https://github.com/Homebrew/homebrew-core/pull/152477
